esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -0,0 +1,205 @@
+        data_domain = self.extra_facets['domain']
+        domain = cx.cordex_domain(data_domain, add_vertices=True)
+        domain_info = cx.domain_info(data_domain)
+        for dim_coord in ['rlat', 'rlon']:
+            old_coord = cube.coord(domain[dim_coord].standard_name)
+            old_coord_dims = old_coord.cube_dims(cube)
+            points = domain[dim_coord].data
+            coord_system = iris.coord_systems.RotatedGeogCS(
+                grid_north_pole_latitude=domain_info['pollat'],
+                grid_north_pole_longitude=domain_info['pollon']
+            )
+            new_coord = iris.coords.DimCoord(
+                points,
+                var_name=dim_coord,
+                standard_name=domain[dim_coord].standard_name,
+                long_name=domain[dim_coord].long_name,
+                units=Unit('degrees'),
+                coord_system=coord_system,
+            )
+            self._check_grid_differences(old_coord, new_coord)
+            new_coord.guess_bounds()
+            cube.remove_coord(old_coord)
+            cube.add_dim_coord(new_coord, old_coord_dims)
+
+    def _fix_geographical_coords(self, cube):
+        """Fix geographical coordinates."""
+        data_domain = self.extra_facets['domain']
+        domain = cx.cordex_domain(data_domain, add_vertices=True)
+        for aux_coord in ['lat', 'lon']:
+            old_coord = cube.coord(domain[aux_coord].standard_name)
+            cube.remove_coord(old_coord)
+            points = domain[aux_coord].data
+            bounds = domain[f'{aux_coord}_vertices'].data
+            new_coord = iris.coords.AuxCoord(
+                points,
+                var_name=aux_coord,
+                standard_name=domain[aux_coord].standard_name,
+                long_name=domain[aux_coord].long_name,
+                units=Unit(domain[aux_coord].units),
+                bounds=bounds
+            )
+            self._check_grid_differences(old_coord, new_coord)
+            aux_coord_dims = (
+                cube.coord(var_name='rlat').cube_dims(cube) +
+                cube.coord(var_name='rlon').cube_dims(cube)
+            )
+            cube.add_aux_coord(
+                new_coord,
+                aux_coord_dims)
+
+    def fix_metadata(self, cubes):
+        """Fix CORDEX rotated grids.
+
+        Set rotated and geographical coordinates to the
+        values given by each domain specification.
+
+        The domain specifications are retrieved from the
+        py-cordex package.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            coord_system = cube.coord_system()
+            if isinstance(coord_system, RotatedGeogCS):
+                self._fix_rotated_coords(cube)
+                self._fix_geographical_coords(cube)
+            if isinstance(coord_system, LambertConformal):
```

Comment:
Perhaps it would be better as an `elif`? In any case, I would suggest also adding an `else` branch to make sure other grids don't just fall through the cracks. This will allows us to search through the logs when someone complains.

Reply:
Ok, will do

Reply:
Done